modules/code-editor/src/views/full/LiteEditor/index.tsx

```diff
@@ -0,0 +1,309 @@
+import _ from 'lodash'
+import * as monaco from 'monaco-editor/esm/vs/editor/editor.api'
+import babylon from 'prettier/parser-babylon'
+import prettier from 'prettier/standalone'
+import React from 'react'
+
+import { FileType } from '../../../backend/typings'
+import { RootStore } from '../store'
+import { END_COMMENT, START_COMMENT, wrapper } from '../utils/wrapper'
+
+import style from './style.scss'
+
+interface Parameters {
+  name: string
+  type: string
+}
+
+interface Props {
+  onChange: (code: string) => void
+  args?: Parameters[]
+  customKey: string
+  code: string
+  maximized: boolean
+  displayed: boolean
+  hints: Hint[]
+  bp: any
+  // Supposed to be string | CustomTemplate but there are some minor issues with common typings & webpack
+  template?: any
+}
+
+interface Hint {
+  scope: 'inputs'
+  name: string
+  source: string
+  category: 'VARIABLES'
+  partial: boolean
+  description?: string
+  location?: string
+  parentObject?: string
+}
+
+function findLastIndex<T>(array: Array<T>, predicate: (value: T, index: number, obj: T[]) => boolean): number {
+  let l = array.length
+  while (l--) {
+    if (predicate(array[l], l, array)) {
+      return l
+    }
+  }
+  return -1
+}
+
+export default class MinimalEditor extends React.Component<Props> {
+  private store: RootStore
+  private editor: monaco.editor.IStandaloneCodeEditor
+  private editorContainer: HTMLDivElement
+
+  constructor(props) {
+    super(props)
+    this.store = new RootStore({ bp: this.props.bp })
+  }
+
+  state = {
+    code: ''
+  }
+
+  componentDidMount() {
+    this.setupEditor()
+
+    // tslint:disable-next-line: no-floating-promises
+    this.loadTypings()
+
+    this.reloadCode(this.props.code)
+    // this.refreshLayout()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.customKey !== this.props.customKey) {
+      this.setState({ code: this.props.code })
+
+      this.loadCodeTypings()
+      this.reloadCode(this.props.code)
+    }
+  }
+
+  componentWillUnmount() {
+    const uri = monaco.Uri.parse('bp://files/index.ts')
```

Comment:
It's required to have one, basically if we implement multi-tabbed files we would need to have a different one per tab. It's static since we only have one